<launch>
  <!-- Load URDF -->
  <param name="robot_description" textfile="$(find pps)/urdf/scanner_housing.urdf"/>

  <!-- Publish TF -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"/>

  <!-- Encoder node -->
  <node pkg="encoder_process" type="encoder_process_node.py" name="encoder_process"/>

  <!-- <node pkg="encoder_process" type="test_runrosbag_laserscan.py" name="test_runrosbag_laserscan"/> -->

  <node pkg="laser_assembler" type="point_cloud2_assembler" name="assembler_node" output="screen">
    <param name="frame_id" value="LMS_511_scan_plane"/>
    <!-- remap should be remove in real implement -->
    <remap from="/cloud" to="/cloud1"/> 
    <param name="fixed_frame" type="string" value="base_link" />
    <param name="max_clouds" type="int" value="3000" />
    <param name="ignore_laser_skew" type="bool" value="false" />
    <param name="downsample_factor" type="int" value="2" />
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pps)/launch/rviz/scanner.rviz"/>

  <node pkg="ui_can_interface" name="ui_can_node" type="ui_can_node.py" />
  <include file="$(find ui)/launch/ui.launch"/>

  <!-- pps package -->
  <node pkg="pps" name="hmi_scan_command_handler" type="hmi_scan_command_handler.py" />




</launch>
